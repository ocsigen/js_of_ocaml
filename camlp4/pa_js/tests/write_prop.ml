(* Test prop writing. *)

fun (obj : int) -> obj##p <- 2;;

fun (obj : < > Js.t) -> obj##p <- 2;;

fun (obj : < p : float Js.readonly_prop > Js.t) -> obj##p <- 1;;

fun (obj : < p : float Js.prop > Js.t) -> obj##p <- 1;;

fun (obj : < p : int Js.prop > Js.t) -> (obj##p <- 1) + 1;;
