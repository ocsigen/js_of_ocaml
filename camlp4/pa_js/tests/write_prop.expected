

#     Characters 45-48:
  fun (obj : int) -> obj##p <- 2;;
                     ^^^
Error: This expression has type int but an expression was expected of type
         < .. > Js.t
#   Characters 25-28:
  fun (obj : < > Js.t) -> obj##p <- 2;;
                          ^^^
Error: This expression has type <  >
       It has no method p
#   Characters 52-58:
  fun (obj : < p : float Js.readonly_prop > Js.t) -> obj##p <- 1;;
                                                     ^^^^^^
Error: This expression has type
         float Js.readonly_prop = < get : float > Js.gen_prop
       but an expression was expected of type
         < set : 'jsoo_arg -> unit; .. > Js.gen_prop
       The first object type has no method set
#   Characters 53-54:
  fun (obj : < p : float Js.prop > Js.t) -> obj##p <- 1;;
                                                      ^
Error: This expression has type int but an expression was expected of type
         float
#   Characters 42-53:
  fun (obj : < p : int Js.prop > Js.t) -> (obj##p <- 1) + 1;;
                                           ^^^^^^^^^^^
Error: This expression has type unit but an expression was expected of type
         int
# 
